node/rpc/src/rpc_util.rs

```diff
@@ -42,6 +47,62 @@ pub fn is_streaming_method(method_name: &str) -> bool {
     STREAMING_METHODS.contains(&method_name)
 }
 
+pub const RPC_METHOD_AUTH_VERIFY: &str = "Filecoin.AuthVerify";
+
+pub async fn check_permissions<DB, KS, B>(
+    rpc_server: JsonRpcServerState,
+    method_name: &str,
+    authorization_header: Option<HeaderValues>,
+) -> Result<(), tide::Error>
+where
+    DB: BlockStore + Send + Sync + 'static,
+    KS: KeyStore + Send + Sync + 'static,
+    B: Beacon + Send + Sync + 'static,
+{
+    match authorization_header {
+        None => return Ok(()),
+        Some(header_values) => match header_values.get(0) {
```

Comment:
We could use [and_then](https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then) to cleanup things here.